### PR TITLE
Changes & instructions to work on Ubuntu 12.04

### PR DESCRIPTION
**Pull request by mrdavidlaing**
Here are some changes & instructions for getting hubot-skype working on a headless Ubuntu 12.04 server.

Hopefully this will address some of the issues that have been raised, at least on the Ubuntu 12.04 platform :)

### CODE REVIEW

File: src/skype.coffee [hunk 1]
```diff
@@ -1,7 +1,8 @@
 Readline = require 'readline'
 
-Robot   = require("hubot").robot()
-Adapter = require("hubot").adapter()
+Robot         = module.parent.parent.exports.Robot
+Adapter       = module.parent.parent.exports.Adapter
+TextMessage   = module.parent.parent.exports.TextMessage
```
Comment (netpro2k) [inline src/skype.coffee:6]: I feel like there must be a more correct way of doing this. Ideas?

File: src/skype.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ def on_message(message, status):
         sys.stdout.write(json_string + '\n')
         sys.stdout.flush()
 
-s = Skype4Py.Skype()
+s = Skype4Py.Skype(Transport='x11')
```
Comment (netpro2k) [inline src/skype.py:17]: This should probably use an environment variable like `$SKYPE_TRANSPORT` or detect based on OS.

File: src/skype.coffee (hunk 1 above)
Comment (mrdavidlaing) [inline src/skype.coffee:6]: Agree - but my coffeescript / nodejs skills don't stretch that far.

Being able to say `Robot   = require("hubot").robot()` seems like the correct syntax, but I could never get that to work.

Could it be something to do with having to include hubot as a dependancy in the package.json?

File: src/skype.py (hunk 2 above)
Comment (mrdavidlaing) [inline src/skype.py:17]: Agreed.  Do you know how to grab an env variable in coffeescript?

**Comment by nicosuave** [comment]
What version of Hubot are you running? I am in the process of configuring this and apparently the adapter format changed, so I run into some issues. I plan on running an old version

**Comment by mrdavidlaing** [comment]
I can't remember exactly, but it was the latest version available around the beginning of Aug 2012 - v2.3.2 I think.

**Comment by netpro2k** [comment]
Have finally gotten around to updating this all so it should now be working on latest version of hubot and install correctly through npm. I also included a different pull request that adds support for setting transport protocol for linux computability.

As for these linux setup instructions i think these would be better suited for a blog post, which I would be happy to link to from the readme.

**Pull request closed without merging**